packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -21,6 +22,62 @@ export function is_hoistable_function(node) {
 	return false;
 }
 
+/**
+ * @param {import("#compiler").SvelteNode[]} nodes
+ */
+function sort_const_tags(nodes) {
+	// If we are in a legacy component, attempt to re-order any ConstTag template nodes
+	// so that they're in topological order given their dependencies.
+	const const_tags = [];
+	const deps = new Set();
+	const tag_ids = new Map();
+	const other = [];
+	for (const node of nodes) {
+		if (node.type === 'ConstTag') {
+			const ids = extract_identifiers(node.declaration.declarations[0].id);
+			for (const id of ids) {
+				deps.add(id.name);
+			}
+			tag_ids.set(node, ids);
+			const_tags.push(node);
+		} else {
+			other.push(node);
+		}
+	}
+
+	if (const_tags.length > 1) {
+		let attempts = 0;
+		const new_nodes = [];
+		while (const_tags.length > 0) {
+			// Worst case-scenario we prevent an infinite loop
+			if (attempts > 100) {
+				new_nodes.push(...const_tags);
+				break;
+			}
```

Comment:
There should be no need to track `attempts` — if there's a cyclical dependency we should detect it properly and error, if we're matching the (frankly insane) Svelte 4 behaviour:

https://svelte.dev/repl/5c791c46bf004b0393f8470e7a952cf6?version=4.2.15

<img width="373" alt="image" src="https://github.com/sveltejs/svelte/assets/1162160/fe2f2ba9-718e-42ab-b0fd-300cab27b8d9">

Reply:
I saw that but wasn't sure how to model it here in the same way. :/